pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/KubernetesRuntimeFactory.java

```diff
@@ -158,6 +164,11 @@ public KubernetesRuntimeFactory(String k8Uri,
         this.expectedMetricsCollectionInterval = expectedMetricsCollectionInterval == null ? -1 : expectedMetricsCollectionInterval;
         this.secretsProviderConfigurator = secretsProviderConfigurator;
         this.functionInstanceMinResources = functionInstanceMinResources;
+        try {
+            setupClient();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Can you please add a logging statement before throwing a RuntimeException?
RuntimeException is very hard to debug.

Reply:
will add